interfaces/builtin/polkit.go

```diff
@@ -0,0 +1,156 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/polkit"
+	"github.com/snapcore/snapd/polkit/validate"
+	"github.com/snapcore/snapd/snap"
+)
+
+const polkitSummary = `allows access to polkitd to check authorisation`
+
+const polkitBaseDeclarationPlugs = `
+  polkit:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const polkitBaseDeclarationSlots = `
+  polkit:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const polkitConnectedPlugAppArmor = `
+# Description: Can talk to polkitd's CheckAuthorization API
+
+#include <abstractions/dbus-strict>
+
+dbus (send)
+    bus=system
+    path="/org/freedesktop/PolicyKit1/Authority"
+    interface="org.freedesktop.PolicyKit1.Authority"
+    member="{,Cancel}CheckAuthorization"
+    peer=(name="org.freedesktop.PolicyKit1", label=unconfined),
+dbus (send)
+    bus=system
+    path="/org/freedesktop/PolicyKit1/Authority"
+    interface="org.freedesktop.DBus.Properties"
+    peer=(name="org.freedesktop.PolicyKit1", label=unconfined),
+dbus (send)
+    bus=system
+    path="/org/freedesktop/PolicyKit1/Authority"
+    interface="org.freedesktop.DBus.Introspectable"
+    member="Introspect"
+    peer=(name="org.freedesktop.PolicyKit1", label=unconfined),
+`
+
+type polkitInterface struct {
+	commonInterface
+}
+
+func (iface *polkitInterface) getActionPrefix(attribs interfaces.Attrer) (string, error) {
+	var prefix string
+	if err := attribs.Attr("action-prefix", &prefix); err != nil {
```

Comment:
I don't know if you be useful to fallback to snap.name prefix in case this is not set?

Reply:
I don't think that is particularly useful, given that actions are intended to be reverse-DNS style names. It is unlikely that this would be a useful default for any existing software, and I'm not sure whether it would be a desirable default for new software either.